invokeai/app/invocations/grounded_sam.py

```diff
@@ -0,0 +1,203 @@
+from pathlib import Path
+from typing import Literal
+
+import numpy as np
+import numpy.typing as npt
+import torch
+from PIL import Image
+from transformers import AutoModelForMaskGeneration, AutoProcessor, pipeline
+from transformers.models.sam import SamModel
+from transformers.models.sam.processing_sam import SamProcessor
+from transformers.pipelines import ZeroShotObjectDetectionPipeline
+
+from invokeai.app.invocations.baseinvocation import BaseInvocation, invocation
+from invokeai.app.invocations.fields import ImageField, InputField
+from invokeai.app.invocations.primitives import ImageOutput
+from invokeai.app.services.shared.invocation_context import InvocationContext
+from invokeai.backend.grounded_sam.detection_result import DetectionResult
+from invokeai.backend.grounded_sam.grounding_dino_pipeline import GroundingDinoPipeline
+from invokeai.backend.grounded_sam.mask_refinement import mask_to_polygon, polygon_to_mask
+from invokeai.backend.grounded_sam.segment_anything_model import SegmentAnythingModel
+
+GROUNDING_DINO_MODEL_ID = "IDEA-Research/grounding-dino-tiny"
+SEGMENT_ANYTHING_MODEL_ID = "facebook/sam-vit-base"
+
+
+@invocation(
+    "grounded_segment_anything",
+    title="Segment Anything (Text Prompt)",
+    tags=["prompt", "segmentation"],
+    category="segmentation",
+    version="1.0.0",
+)
```

Comment:
Do we want to flag this as beta for now? I don't mind either way, just checking.

Reply:
@psychedelicious Think its a safe enough implementation that we don't need beta for it. At least as far as the node is concerned. As for the real implementation of this, I think that needs some UI work.